src/tallies/tally_filter_cell.F90

```diff
@@ -118,4 +119,28 @@ function text_label_cell(this, bin) result(label)
     label = "Cell " // to_str(cells(this % cells(bin)) % id)
   end function text_label_cell
 
+!===============================================================================
+!                               C API FUNCTIONS
+!===============================================================================
+
+  function openmc_cell_filter_get_bins(index, cells, n) result(err) bind(C)
+    ! Return the cells associated with a cell filter
+    integer(C_INT32_T), value :: index
+    type(C_PTR), intent(out) :: cells
+    integer(C_INT32_T), intent(out) :: n
+    integer(C_INT) :: err
+
+    err = verify_filter(index)
+    if (err == 0) then
+       select type (f => filters(index) % obj)
+       type is (CellFilter)
+          cells = C_LOC(f % cells)
```

Comment:
We should probably make `CellFilter % cells` be declared as `integer(C_INT32_T)`.